WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/adapters/ScanAdapter.kt

```diff
@@ -0,0 +1,64 @@
+package org.wordpress.android.ui.jetpack.scan.adapters
+
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.RecyclerView.Adapter
+import org.wordpress.android.ui.jetpack.scan.ScanListItemState
+import org.wordpress.android.ui.jetpack.scan.ScanListItemState.ScanState
+import org.wordpress.android.ui.jetpack.scan.ScanListItemState.ViewType
+import org.wordpress.android.ui.jetpack.scan.adapters.viewholders.ScanStateViewHolder
+import org.wordpress.android.ui.jetpack.scan.adapters.viewholders.ScanViewHolder
+import org.wordpress.android.ui.utils.UiHelpers
+
+class ScanAdapter(private val uiHelpers: UiHelpers) : Adapter<ScanViewHolder>() {
+    private val items = mutableListOf<ScanListItemState>()
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ScanViewHolder {
+        return when (viewType) {
+            ViewType.SCAN_STATE.id -> ScanStateViewHolder(uiHelpers, parent)
+            else -> throw IllegalArgumentException("Unexpected view type in ${this::class.java.simpleName}")
+        }
+    }
+
+    override fun onBindViewHolder(holder: ScanViewHolder, position: Int) {
+        holder.onBind(items[position])
+    }
+
+    override fun getItemViewType(position: Int) = items[position].type.id
+
+    override fun getItemId(position: Int): Long = items[position].longId()
+
+    override fun getItemCount() = items.size
+
+    fun update(newItems: List<ScanListItemState>) {
+        val diffResult = DiffUtil.calculateDiff(ScanDiffCallback(this.items.toList(), items))
+        items.clear()
+        items.addAll(newItems)
+
+        diffResult.dispatchUpdatesTo(this)
+    }
+
+    class ScanDiffCallback(
+        private val oldList: List<ScanListItemState>,
+        private val newList: List<ScanListItemState>
+    ) : DiffUtil.Callback() {
+        override fun areItemsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
+            val oldItem = oldList[oldItemPosition]
+            val newItem = newList[newItemPosition]
+            return when {
+                oldItem is ScanState && newItem is ScanState -> // TODO: ashiagr revisit
```

Comment:
Question (❓ ): What would you revisit here?

Reply:
The following line that compares state title and description to find if state view items are the same. There might be more states in the future which may have same title and description.

Reply:
Got it, thanks for sharing! 🌟